storage.go

```diff
@@ -3,34 +3,297 @@ package minio
 import (
 	"context"
 	"io"
+	"path/filepath"
+	"strings"
 
+	"github.com/minio/minio-go/v7"
+
+	"github.com/beyondstorage/go-storage/v4/pkg/iowrap"
+	"github.com/beyondstorage/go-storage/v4/services"
 	. "github.com/beyondstorage/go-storage/v4/types"
 )
 
+const defaultListBufferSize = 200
+
 func (s *Storage) create(path string, opt pairStorageCreate) (o *Object) {
-	panic("not implemented")
+	o = s.newObject(false)
+	o.Mode = ModeRead
+	o.ID = s.getAbsPath(path)
+	o.Path = path
+
+	return o
 }
 
 func (s *Storage) delete(ctx context.Context, path string, opt pairStorageDelete) (err error) {
-	panic("not implemented")
+	var fileVersion string
+	if opt.HasFileVersion {
+		fileVersion = opt.FileVersion
+	}
+
+	objectType := "file"
+	if opt.HasObjectType {
+		objectType = opt.ObjectType
+	}
+
+	uniquePath := s.getAbsPath(path)
+	switch objectType {
+	case "file":
+		err = s.client.RemoveObject(ctx, s.bucketName, uniquePath, minio.RemoveObjectOptions{
+			GovernanceBypass: true,
+			VersionID:        fileVersion,
+		})
+		if err != nil {
+			return err
+		}
+	case "directory":
+		ctxWithCancel, cancel := context.WithCancel(ctx)
+		defer func() {
+			// ctxWithCancel will return none-nil if closed/cancelled ...
+			if ctxWithCancel.Err() == nil {
+				cancel()
+			}
+		}()
+
+		// minio couldn't stat/delete dir object directly
+		// ref: https://github.com/minio/minio-go/issues/803
+		// using list to found
+		findChan := s.client.ListObjects(ctxWithCancel, s.bucketName, minio.ListObjectsOptions{
```

Comment:
`Delete` only delete one object. We don't need support to delete a dir.

Reply:
Comment for `delete` updated here: https://github.com/beyondstorage/specs/pull/92